subsys/net/ip/ipv4.c

```diff
@@ -139,6 +139,10 @@ int net_ipv4_parse_hdr_options(struct net_pkt *pkt,
 			return -EINVAL;
 		}
 
+		if (opts_len < 1U) {
```

Comment:
This looks weird. unsiged value less than 1 is unsigned value of exactly 0. And header of the loop is `while (opts_len)`, and opts_len is not modified before this `if`. So, it would never execute.

Perhaps, the condition should be `<=`, or actually, the check should be moved after `opts_len--` (and be against 0).

Reply:
yeah, not getting it as well, the `while (opts_len)` covers the case where it is 0 and `opts_len` does not change between that check and this `if`.

Reply:
Yep, removed.